org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractContentAssistTest.java

```diff
@@ -104,4 +124,132 @@ protected void initializeTypeProvider(XtextResourceSet resourceSet) {
 		resourceSet.setClasspathURIContext(getJavaProject(resourceSet));
 	}
 
+	/**
+	 * Creates a dsl file.
+	 * 
+	 * @param fileName
+	 * 		The name of the file (without the file extension).
+	 * 		To be able to specify the workspace relative path,
+	 * 		use {@link IResourcesSetupUtil#createFile(String, String)} instead.
+	 * 
+	 * @param content
+	 * 		The content of the file.
+	 * 
+	 * @since 2.35
+	 */
+	protected IFile createDslFile(String fileName, CharSequence content) {
+		return createDslFile(fileName, fileExtensionProvider.getPrimaryFileExtension(), content);
+	}
+
+	/**
+	 * Creates a dsl file.
+	 * 
+	 * @param fileName
+	 * 		The name of the file (without the file extension)
+	 * 
+	 * @param fileExtension
+	 * 		The extension of the file.
+	 * 
+	 * @param content
+	 * 		The content of the file.
+	 * 
+	 * @since 2.35
+	 */
+	protected IFile createDslFile(String fileName, String fileExtension, CharSequence content) {
+		try {
+			return IResourcesSetupUtil.createFile(javaProject.getElementName(), fileName, fileExtension, content.toString());
+		} catch (InvocationTargetException | CoreException | InterruptedException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	/**
+	 * Verifies whether the content assistant provides the expected proposals on the given cursor position.
+	 * 
+	 * @param text
+	 * 		The editor's input text. The text must contain the {@link #c} special symbol indicating the current cursor position.
+	 * 
+	 * @param expectedProposals
+	 * 		The proposals (separated by new lines) that are expected to be offered whenever the content assistant is triggered on the given cursor position.
+	 * 
+	 * @since 2.35
+	 */
+	protected void assertContentAssistant(CharSequence text, CharSequence expectedProposals) {
+		assertContentAssistant(text, expectedProposals.toString().split(System.lineSeparator()), null, null);
+	}
+	
+	/**
+	 * Verifies whether the content assistant provides the expected proposals on the given cursor position.
+	 * 
+	 * @param text
+	 * 		The editor's input text. The text must contain the {@link #c} special symbol indicating the current cursor position.
+	 * 
+	 * @param expectedProposals
+	 * 		The proposals that are expected to be offered whenever the content assistant is triggered on the given cursor position.
+	 * 
+	 * @since 2.35
+	 */
+	protected void assertContentAssistant(CharSequence text, String[] expectedProposals) {
```

Comment:
So you removed the version with List?

That could have been useful in Xtend with `#[ ... ]`. Or would it work anyway thanks to conversion?

Moreover, if we want to use arrays, why not using `String... expectedProposals`, which would not require the creation of an array when calling this method? Would it conflict with the one accepting a String?